internal/tools/crosslink/crosslink.go

```diff
@@ -0,0 +1,170 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The crosslink tool generates and maintains replace directives in all
+// the go.mod files within this repository. Some directives are superfluous
+// (e.g. because the replaced module doesn't occur in the dependency tree),
+// but we generate them anyway for the sake of consistency.
+//
+// In particular, we generate a replace directive from each module to itself
+// (i.e., the target path "./"). This is actually necessary in the presence of
+// cyclic dependencies between modules.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/tabwriter"
+)
+
+type repo string
+
+func findRepoRoot() (repo, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	for {
+		_, err := os.Stat(filepath.Join(dir, ".git"))
+		if errors.Is(err, os.ErrNotExist) {
+			dir = filepath.Dir(dir)
```

Comment:
Can we add a check to ensure there isn't an infinite loop if `.git` is not found and this backs all the way out to the root?

```suggestion
			dir = filepath.Dir(dir)
			if dir == string(os.PathSeparator) {
				return "", err
			}
```

Reply:
I agree with doing this, but I need to figure out how. The suggested approach wouldn't work on Windows (because root file paths probably look like `C:\` rather than `\`).

Reply:
Done, using a slight variation on the proposal. Thanks!